### PR TITLE
workflow: update dependencies

### PR DESCRIPTION
**Pull request by jonchang**
Testing for workflow.

### CODE REVIEW

**Comment by jonchang** [comment]
Vendored boost fails on macOS building in c++17 mode with clang.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -24,18 +24,23 @@ jobs:
...
         fi
         python3 -mpip install meson ninja
     - name: Build
       run: |
         meson build --prefix=$HOME/local --buildtype=release
-        ninja -C build install
+        ninja -C build install -j4
     - name: Test
       run: |
+        export PATH=$HOME/local/bin
```
Comment (jonchang) [inline .github/workflows/build.yml]:
```suggestion
        export PATH=$HOME/local/bin
        export PATH=$HOME/local/bin:$PATH
```

**Comment by jonchang** [comment]
> simple typo here: should be `export PATH=$HOME/bin:$PATH` - but you knew that
> 
> also this is fun. I have never used this commenting tool before

You can also make suggestions inline!

**Comment by jonchang** [comment]
> For doing `ninja -j4`, my impression is that ninja auto-discovers how many processes it should run. This could still maybe speed things up, if the virtual machine is configured to look like it has only one core.

Yeah this is the case. The documentation says "2 cores only" but from the grapevine they say that the machines are actually far more powerful than they advertise.

**Comment by jonchang** [comment]
Squashed and rebased given the boost changes.

**Comment by bredelings** [comment]
BTW, I'm not really sure what's happening with boost.  Debian and ubuntu packages don't have these weird -mt-x64 suffixes on the end.  I think that madness is mostly inspired by Windows systems, which have e.g. different versions of the libraries (!) if its a debug build or not. 

It looks like maybe it was finding some non-system version of boost, but only during compilation.  Then, when attempting to actually RUN the programs, it could only find the system version.

**Comment by jonchang** [comment]
> It looks like maybe it was finding some non-system version of boost, but only during compilation. Then, when attempting to actually RUN the programs, it could only find the system version.

Yup, GHA runners installs some voodoo version of boost which was causing trouble: https://github.com/actions/virtual-environments/blob/master/images/linux/scripts/installers/boost.sh

**Comment by bredelings** [comment]
That's gross.  `rm -rf` indeed!

**Comment by jonchang** [comment]
I feel like this is the last stage of the war with CI.

Next step is to set up a release pipeline!

**Comment by jonchang** [comment]
This is ready to merge. Feel free to `git rm .travis.yml` whenever 👋

**Comment by bredelings** [comment]
Thanks Jon, that is awesome!  I am away from wifi until tomorrow morning but will happily remove Travis then.

**Comment by bredelings** [comment]
Worka..

**Pull request merged**